website/docs/docs-section/roadmap/create-custom-context.md

```diff
@@ -4,4 +4,181 @@ title: Creating Custom Context
+        "city": "Singapore",
+        "postalCode": "612345",
+        "phoneNumber": "61204028"
+      },
+      "name": "James Lee",
+      "email": "def@company.com"
+    }
+    // other properties
+  }
+}
+```
+
+The `.json` above shows the structure of the `Invoice`'s raw document. The context created earlier, can be seen to be included in the `@context` field as a URL link.
+
+During the [wrapping](/docs/developer-section/libraries/open-attestation-cli#wrapping-documents) process, the document itself as well as the fields inside `credentialSubject` would be validated and cross checked against the contexts defined in `@context`.
+
+:::note
+It is not necessary that everything in `@context` has to appear in `credentialSubject`. But when there are additional items in `credentialSubject` that do not exist in what has already been defined in `@context`, an error would occur when trying to wrap the document.
+:::
+
+## Configuration File (Invoice)
+
+```json
+{
+  "network": "ropsten",
+  "wallet": {
+    "type": "ENCRYPTED_JSON",
+    "encryptedJson": "{\"address\":\"1245e5b64d785b25057f7438f715f4aa5d965733\",\"id\":\"bf069d1b-4e88-487c-b695-f2e03ed7c1ff\",\"version\":3, ...}"
+  },
+  "forms": [
+    {
+      "name": "Invoice",
+      "type": "VERIFIABLE_DOCUMENT",
+      "defaults": {
+        "version": "https://schema.openattestation.com/3.0/schema.json",
+        "@context": [
+          "https://www.w3.org/2018/credentials/v1",
+          "https://schemata.openattestation.com/io/tradetrust/invoice/1.0/invoice-context.json",
+          "https://schemata.openattestation.com/com/openattestation/1.0/OpenAttestation.v3.json"
+        ],
+        "type": ["VerifiableCredential", "OpenAttestationCredential"],
+        "issuanceDate": "2010-01-01T19:23:24Z",
+        "issuer": {
+          "id": "https://example.com",
+          "name": "DEMO STORE",
+          "type": "OpenAttestationIssuer"
+        },
+        "openAttestationMetadata": {...},
+        "credentialSubject": {}
+      },
+      "schema": {...}
+    }
+  ]
+}
+```
+
+The `.json` above shows the structure of the `Invoice`'s configuration file. Apart from the rest of the information such as `schema`, that contains custom fields to be filled in during the creation of the document, the `@context` could also be seen to be included.
+
+Similarly, the items inside `@context` contain the types of what are the different fields required in the document. It would be used to cross check against the items that have been filled in inside `schema` to ensure that the document created using this configuration file does not contain anything more than what has already been defined in `@context`.
+
+:::note
+It is not necessary that everything in `@context` has to appear in `schema`. But when there are additional items in `schema` that do not exist in what has already been defined in `@context`, an error would occur when trying to create a document.
```

Comment:
I don't know why but I suddenly have a question that may sound silly.😅 What is the difference between a context and schema, and what differences or observations should I expect when there are custom fields in the `context` that don't exist in the `schema`?

Reply:
`Schema`: used by both v2 and v3 to validate the structure of the document when wrapping.
`Context`: used only in v3 to provide globally unique identifiers for standard properties and types used to describe data. <-- copied the definition haha. So the context basically describes what has been defined in the URI that the fields of the document should have. Like the fields of the invoice in this case.

I would say the schema is to validate the fields in the document similar to @context. But with @context, it's now made compulsory to validate the stuff in v3 docs.

Reply:
Json of is about semantic, about the meaning of the fiel data

Reply:
Thanks guys!! So here's a summary of what I've understood about `schema` and `context` in OA. Please correct me if any of the points I've mentioned are incorrect.😬

* Schema is only used in v2 of OA.
* Schema is used during the pre-wrapping process by the CLI v2 by passing a JSON file to the `--schema` option to validate the data in a doc.
* The validation is not used anywhere else in OA after the wrapping process. This applies to both v2 and v3.
* A context does the job of describing and the possibility to use it for validating data in a doc. But only v3 recognises context in a doc.
* Hence, in v3, there is no need for schema because a context is sufficient to define the data and can be used for performing validation during the pre-wrapping process of a doc
* In v3, the `--schema` option in the CLI will not work as that is only meant for v2.
* The context array will be compulsory in a v3 doc.

Reply:
I didnt read all your discussion, but:
- there is a schema for oa v2 and v3. Schemata are optional (not enforced) but recommended and can be used for v2 and v3
- Yes
- Yes and there is a reason, can you find it ?
- Context is not meant for data validation. It's just [semantic web](https://en.wikipedia.org/wiki/Semantic_Web)
- Nope see above
- Nope see above
- Yes but it's the VC spec, not our decision

Reply:
Cool! Thanks for making it much clearer!😃